### PR TITLE
Test: Add replay block command for nakamoto blocks

### PR DESCRIPTION
**Pull request by kantai**
### Description

This replicates the replay-block functionality in `stacks-inspect` for nakamoto blocks. It also adds a flag for the command to switch between testnet and mainnet configs.

### CODE REVIEW

**Review by obycode — DISMISSED** [review]
LGTM!

File: stackslib/src/cli.rs [hunk 1]
```diff
@@ -68,6 +69,44 @@ impl StacksChainConfig {
...
+        let mut pox_constants = PoxConstants::regtest_default();
+        pox_constants.prepare_length = 100;
+        pox_constants.reward_cycle_length = 900;
+        pox_constants.v1_unlock_height = 3;
+        pox_constants.v2_unlock_height = 5;
+        pox_constants.pox_3_activation_height = 5;
+        pox_constants.pox_4_activation_height = 6;
+        pox_constants.v3_unlock_height = 7;
+        let mut epochs = STACKS_EPOCHS_REGTEST.to_vec();
+        epochs[0].start_height = 0;
```
Comment (obycode) [inline stackslib/src/cli.rs:83]: At some point, I had started making a change that added an `index()` to `StacksEpochId`. I couldn't decide if I liked it or not... what do you think? Would that be helpful for places like this? So for example, here you could write:
```rust
epochs[StacksEpochId::Epoch20.index()].start_height = 0;
```

File: stackslib/src/chainstate/nakamoto/mod.rs [hunk 2]
```diff
@@ -4275,6 +4278,46 @@ impl NakamotoChainState {
             .as_ref()
             .map(|rewards| rewards.reward_info.clone());
 
+        if do_not_advance {
+            // get burn block stats, for the transaction receipt
+            let (parent_burn_block_hash, parent_burn_block_height, parent_burn_block_timestamp) =
+                if block.is_first_mined() {
+                    (BurnchainHeaderHash([0; 32]), 0, 0)
```
Comment (jcnelson) [inline stackslib/src/chainstate/nakamoto/mod.rs:4285]: Is this ever reachable?

File: stackslib/src/chainstate/nakamoto/mod.rs [hunk 3]
```diff
@@ -4275,6 +4278,46 @@ impl NakamotoChainState {
             .as_ref()
             .map(|rewards| rewards.reward_info.clone());
 
+        if do_not_advance {
```
Comment (jcnelson) [inline stackslib/src/chainstate/nakamoto/mod.rs:4281]: Can this code block be its own method?  This function is already huge.

File: stackslib/src/main.rs [hunk 4]
```diff
@@ -1470,6 +1470,35 @@ simulating a miner.
...
 
+    if argv[1] == "replay-naka-block" {
+        let chain_config =
+            if let Some(network_flag_ix) = argv.iter().position(|arg| arg == "--network") {
+                let Some(network_choice) = argv.get(network_flag_ix + 1) else {
+                    eprintln!("Must supply network choice after `--network` option");
+                    process::exit(1);
+                };
+
+                let network_config = match network_choice.as_str() {
```
Comment (jferrant) [inline stackslib/src/main.rs:1481]: This should probably be case insensitive no?

Comment (kantai) [reply stackslib/src/main.rs]: Sure -- updated in https://github.com/stacks-network/stacks-core/pull/5346/commits/84950edfa160e6662b3cbd41a14522fcf2fb9ba7

File: stackslib/src/chainstate/nakamoto/mod.rs (hunk 3 above)
Comment (kantai) [reply stackslib/src/chainstate/nakamoto/mod.rs:4335]: Makes sense, updated in https://github.com/stacks-network/stacks-core/pull/5346/commits/84950edfa160e6662b3cbd41a14522fcf2fb9ba7

File: stackslib/src/chainstate/nakamoto/mod.rs (hunk 2 above)
Comment (kantai) [reply stackslib/src/chainstate/nakamoto/mod.rs]: Nope -- removed in https://github.com/stacks-network/stacks-core/pull/5346/commits/84950edfa160e6662b3cbd41a14522fcf2fb9ba7

**Review by obycode — APPROVED** [review]
LGTM

**Review by jferrant — DISMISSED** [review]

**Review by obycode — DISMISSED** [review]

**Review by obycode — DISMISSED** [review]

**Review by wileyj — APPROVED** [review]

**Review by jcnelson — APPROVED** [review]

**Pull request merged**